iroh-net/src/magicsock.rs

```diff
@@ -717,6 +732,31 @@ impl AsyncUdpSocket for MagicSock {
 
         let mut num_msgs = 0;
         for (buf_out, meta_out) in bufs.iter_mut().zip(metas.iter_mut()) {
+            // first read from self.inner.cached_recv_datagrams
+            {
+                if self.inner.is_closed() {
+                    break;
+                }
+                let mut cached = self.inner.cached_recv_datagrams.lock().unwrap();
+                if let Some((mut meta, mut bytes)) = cached.take() {
+                    let copy_len = if bytes.len() > buf_out.len() {
+                        let segment_count = buf_out.len() / meta.stride;
```

Comment:
Unfortunately the docs about RecvMeta are rather silent about the meaning of stride.

https://docs.rs/quinn-udp/latest/quinn_udp/struct.RecvMeta.html

But is this guaranteed to be a whole number, or could you have a situation where e.g. buf is 1500 bytes and stride is 1200 bytes, so you get 1 1200 byte packet and 1 300 byte packet?

At least on the send side I saw something like this - segment_size being 1200 and the total size being something that is not a multiple of the segment_size.

See the log in https://github.com/n0-computer/iroh/issues/1357#issuecomment-1679110798

Reply:
I implemented the "split on send" thing here: https://github.com/n0-computer/iroh/pull/1380

Reply:
So RecvMeta docs are updated in the quinn main branch.

Your understanding is correct: the last datagram could be smaller than the stride/segment-size (i'm not sure why they call this something different on the send and receive paths).